### ISSUE TITLE
Fix Bugs in Kubeconfig Sync

### ISSUE THREAD

**Issue by Nokel81** [lensapp/lens#2663]
**Describe the bug**
Once #2567 lands there will be the following items that need to fixed:

- [x] Remove notifcations for invalid contexts.
- [x] Fix catalog being empty after closing and opening window (#2789)
- [x] Change all delete clusters to run `kubectl config delete context --kubeconfig <path>` (#2923)
- [ ] #3439
- [ ] #3438
- [ ] #3437
- [ ] #3436
- [ ] ~~refactor cluster to store the copy (in memory) of the KubeConfig itself so that we both don’t read from the FS as often and so we can do diffing~~
- [x] Find some way to produce consistent IDs so that settings work
- [x] Remove kube config file picker
- [ ] #3435
- [x] Investigate where AWS, GKE, store their configs. So that we can "auto-detect" those clusters. (These seem to be just added to the default location)

**Comment by jakolehm** [comment]
Couple more:

- ~~sync does not watch internal (c&p) folder at all (imho **blocker**)~~
- it's possible to remove default kubeconfig path, bug or feature?
- Preferences say "Sync an individual file or all files in a folder (non-recursive)." although scan _IS_ recursive.

**Comment by Nokel81** [comment]
- Second was feature, see slack discussion.
- If that is the case then it is a bug, I though I checked to make sure that it wasn't recursive.

**Comment by Nokel81** [comment]
Closing in favour of the subtickets.
